### PR TITLE
Weather APP

### PR DESCRIPTION
**Pull request by rachaouldbabaali**
In this PR, I have :

 

- [x] Developed reusable components including Navbar, Details, City, and Heading.

- [x] Implemented a Home page that has a search bar with autosuggestion.

- [x] Developed a cities page that dynamically loads a list of cities based on the user's selected country.

- [x] Implemented a Details page that fetches weather data from an API and displays it to the user.

- [x] Styled the entire application to meet the design need based on the Ballhead App design idea by Nelson Sakwa on Behance.

link to the presentation [vidéo](https://www.loom.com/share/09fd4aa765cd4956961fadd273aa970f)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,2 +1,168 @@
...
+## ⭐️ Show your support <a name="support"></a>
+
+If you like this project please give it a start and share it with your friends.
+
+<p align="right">(<a href="#readme-top">back to top</a>)</p>
+
+## 🙏 Acknowledgments <a name="acknowledgements"></a>
+
+I would like to appreciate[Microverse](https://www.microverse.org/) for providing the resources and the platform for us to be a Professional full-stack developer.
+
```
Comment (omarsalem7) [inline README.md]: - [ ] Kindly consider giving appropriate credit to [Nelson Sakwa](https://www.behance.net/sakwadesignstudio) the [author of the original design](https://www.behance.net/gallery/31579789/Ballhead-App-(Free-PSDs)), as required by the [Creative Commons License](https://creativecommons.org/licenses/).

File: README.md [hunk 2]
```diff
@@ -1,2 +1,168 @@
...
+The app consists of three pages:
+
+- **The Home page displays a list of cities and the weather there.**
+- **The Details page displays the weather details for the selected city.**
+
+<p align="right">(<a href="#readme-top">back to top</a>)</p>
+
+## 🚀 Live Demo <a name="live-demo"></a>
+
+[N/A]
```
Comment (omarsalem7) [inline README.md]: - [ ] Kindly consider deploying the application using Render, Netlify or GitHub Pages.

**Review by omarsalem7 — CHANGES_REQUESTED** [review]
Hi @rachaouldbabaali 😊, 

Good job so far! 👏
You have done well reaching this point of the milestone, but there are some issues that you still need to work on to go to the next project but you are almost there! 

you-are-near

## Highlights
- App created using react app ✔️
- Use react-router properly✔️

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
- N/A

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by rachaouldbabaali** [comment]
Thank you @omarsalem7 😃

**Review by Ismailco — APPROVED** [review]
### Hi @rachaouldbabaali, 

In my opinion, your project is ready for presentation!  There is nothing else to say other than... it's time to merge it :shipit: 

cr-approve

## Highlights

✔️ Follow design guidelines
✔️ No linter errors
✔️ Used Gitflow
✔️ Work documented professionally

Good luck! 🎉 

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**Comment by rachaouldbabaali** [comment]
thank you @Ismailco

**Pull request merged**